llvm/lib/Transforms/IPO/WholeProgramDevirt.cpp

```diff
@@ -1533,11 +1544,26 @@ void DevirtModule::applyICallBranchFunnel(VTableSlotInfo &SlotInfo,
       llvm::append_range(Args, CB.args());
 
       CallBase *NewCS = nullptr;
+      if (!JT.isDeclaration() && !ProfcheckDisableMetadataFixes) {
```

Comment:
Can you add some comments about the approach used below to compute the entry counts?

Reply:
ptal